src/components/DeleteProbeButton/DeleteProbeButton.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { Button, Tooltip } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import type { BackendError, DeleteProbeButtonProps } from './DeleteProbeButton.types';
+import { useDeleteProbe } from 'data/useProbes';
+import { useCanEditProbe } from 'hooks/useCanEditProbe';
+import { ConfirmModal } from 'components/ConfirmModal';
+import { ProbeUsageLink } from 'components/ProbeUsageLink';
+
+import { getPrettyError } from './DeleteProbeButton.utils';
+
+export function DeleteProbeButton({ probe }: DeleteProbeButtonProps) {
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const onDeleteSuccess = useCallback(() => {
+    setShowDeleteModal(false);
```

Comment:
Can we handle the redirect back to the Probes list here explicitly?

I don't like it gets handled by the `useEffect` in `EditProbe` as there is a render in-between which threw me off somewhat.

Reply:
I've added a `onDeleteSuccess` (optional) prop to the `DeleteProbeButton`.
It will now navigate on delete (the `useEffect` has not been removed as still has a purpose).